eng/Analyzers_ShippingRules.ruleset

```diff
@@ -22,4 +22,19 @@
     <Rule Id="RS1022" Action="None" />
   </Rules>
 
+  <Rules AnalyzerId="Roslyn.Diagnostics.Analyzers" RuleNamespace="Roslyn.Diagnostics.Analyzers">
+    <!-- Do not use generic 'CodeAction.Create' to create 'CodeAction' -->
+    <Rule Id="RS0005" Action="None" />
+    <!-- Parts exported with MEFv2 must be marked with 'SharedAttribute' -->
+    <Rule Id="RS0023" Action="None" />
```

Comment:
Roslyn uses MEF for declaring CodeFixProvider and RefactoringProvider, but it doesn't use MEF for constructing them. The `[Shared]` attribute is somewhere between meaningless and misleading because it will still create multiple instances in some cases. I'd leave the rule disabled.